src/catalog/data/slice.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+
+export const initialCatalogState = () => ({
+  fetching: false,
+  errors: [],
+  courses: [],
+});
+
+export const baseCatalogReducers = {
+  catalogSetFetching(state, { payload }) {
+    const { fetching, errors } = payload;
+    state.fetching = fetching;
+    state.errors = errors;
+  },
+  catalogSetCourses(state, { payload }) {
+    state.courses = payload.courses;
+  },
+};
```

Comment:
This seems like a great approach, but what I've generally seen in these MFEs is a separate stage for each of these states, for example: fetchCatalogRequest, fetchCatalogSuccess, fetchCatalogFailure, fetchCatalogDenied. 

I don't think there is anything wrong with the above code, but it's not the pattens people will see in MFEs.